sstables/sstables_manager.cc

```diff
@@ -169,6 +169,41 @@ void sstables_manager::increment_total_reclaimable_memory_and_maybe_reclaim(ssta
     smlogger.info("Reclaimed {} bytes of memory from SSTable components. Total memory reclaimed so far is {} bytes", memory_reclaimed, _total_memory_reclaimed);
 }
 
+future<> sstables_manager::maybe_reload_reclaimed_components() {
+    size_t memory_reclaim_threshold = _available_memory * _db_config.components_memory_reclaim_threshold();
+    size_t memory_available_for_reclaimable_components = memory_reclaim_threshold - _total_reclaimable_memory;
+
+    // Reload bloom filters from the smallest to largest so as to maximize
+    // the number of bloom filters being reloaded.
+    while (!_reclaimed.empty() && memory_available_for_reclaimable_components > 0) {
+        auto sstable_to_reload = _reclaimed.begin();
+        const size_t reclaimed_memory = sstable_to_reload->total_memory_reclaimed();
+        if (reclaimed_memory > memory_available_for_reclaimable_components) {
+            // cannot reload anymore sstables
+            break;
+        }
+
+        // Increment the total memory before reloading to prevent any parallel
+        // fibers from loading new bloom filters into memory.
+        _total_reclaimable_memory += reclaimed_memory;
+        _reclaimed.erase(sstable_to_reload);
+        // Use a lw_shared_ptr to prevent the sstable from getting deleted when
+        // the components are being reloaded.
+        auto sstable_ptr = sstable_to_reload->shared_from_this();
+        try {
+            memory_available_for_reclaimable_components -= co_await sstable_ptr->reload_reclaimed_components();
+        } catch (std::exception& e) {
```

Comment:
Better: catch (...) to not rely on inheritance from std::exception (and use std::current_exception())

Reply:
Done.